### DISCUSSION TITLE
Isolating labelled frames

### DISCUSSION THREAD

**Discussion by rpartan** [talmolab/sleap#2032] [Help!]
I am trying to train a model on very large videos (over 1,000,000 frames). The training never got past Epoch 1 because the computer does not have enough memory to run the whole video. Is there a way I can isolate just the frames I have already labelled to start training?

Thanks,
RP

**Comment by gitttt-1234** [comment]
Hi @rpartan,

There are a couple of options here:
 1. If you know the indices of the labelled frames, then you can explicitly specify the indices for training and validation split in the `training_config.json` file, as given below. However, this option is only available with CLI (and not GUI). Related issue: #1962 
 ```
        "labels": {
            "training_labels": null,
            "validation_labels": null,
            "validation_fraction": 0.1,
            "test_labels": null,
            "split_by_inds": false,
            "training_inds": [2,3,19,24,25,31,34,41,45,53,56,60,...],
            "validation_inds": null,
            "test_inds": null,
            "search_path_hints": [],
            "skeletons": []
        },
```
 2. If you don't have the list of indices, then you can get a new labels file by isolating `user_labelled_instances` with the below code, using [`sleap_io`](https://github.com/talmolab/sleap-io):
 ```py
import sleap_io as sio

labels = sio.load_slp("train.slp")
filtered_lfs = []
for lf in self.labels:
    if lf.user_instances is not None and len(lf.user_instances) > 0:
        lf.instances = lf.user_instances
        filtered_lfs.append(lf)
new_labels = sio.Labels(
    videos=labels.videos,
    skeletons=labels.skeletons,
    labeled_frames=filtered_lfs,
)
new_labels.save("train_labels.slp", embed=True)

```
The new `train_labels.slp` will only contain frames with user-labelled instances.

Let me know if you have any questions!

Thanks,

Divya
